cmd/docker/docker.go

```diff
@@ -259,7 +260,18 @@ func runDocker(dockerCli *command.DockerCli) error {
 	// We've parsed global args already, so reset args to those
 	// which remain.
 	cmd.SetArgs(args)
-	return cmd.Execute()
+	err = cmd.Execute()
+	if err != nil {
+		return err
+	}
+
+	// If the command is being executed in an interactive terminal,
+	// run the plugin hooks (but don't throw an error if something misbehaves)
+	if term.IsTerminal(dockerCli.Out().FD()) {
+		_ = pluginmanager.RunPluginHooks(dockerCli, cmd, args)
+	}
+
+	return nil
```

Comment:
Could we also pass an error to the hooks? This could allow hooks to react to known/common user errors and suggest fixes or correct usage.

Reply:
Oh, I didn't think about that but it's a good idea! Will have to investigate a little bit to see how much information we pass in the errors though – passing a lot of information about the command being executed into the plugin is part of what I was trying to avoid with the "templating" approach.

Reply:
thoughts @thaJeztah @eunomie ?

Reply:
Oh! Thought I replied to this one;

Yes, I think it could make sense to provide the hook access to that information. I can indeed see cases where (potentially) a hook could enrich errors as well, or it would allow experimenting with an out-of-band binary to improve error messages (and provide hints).

One thing I'm wondering is if we should change the signature of this function to accept some struct as last argument; that struct could contain "optional" information, which would allow adding more "context" in future, without having to change the signature.

Also considering if we need to pass a `context.Context` in the new functions; wdyt?

Reply:
I just noticed (some debugging) that we don't call hooks after plugin commands (?) was that intentional (i.e., is it expected for plugins to call hooks?) or something we overlooked?

Reply:
> One thing I'm wondering is if we should change the signature of this function to accept some struct as last argument; that struct could contain "optional" information, which would allow adding more "context" in future, without having to change the signature.

Could you provide an example? Re: `pluginmanager.RunPluginHooks`, I'm not sure what we would pass in for "optional" information right now, and if we need it in the future we can just change the signature, right? This is a function that we're calling from inside the CLI, not from plugins, so changing it's signature shouldn't be an ordeal.

> Also considering if we need to pass a context.Context in the new functions; wdyt?

I'm not sure. If we're talking about `pluginmanager.RunPluginHooks`, for now, the benefits it brings are limited – we don't need it inside, and we don't need it to be able to cancel execution, since it's getting called synchronously. However, if we have a misbehaving plugin, there might be some benefit to passing the context in so that we can pass it to the plugin command being executed to make it easier to kill it? Merits some looking into I guess.

Reply:
> I'm not sure what we would pass in for "optional" information right now,

Yeah, perhaps not needed yet. I noticed the `args` and wondered what other information we may want to pass (and could already have available).

> for now, the benefits it brings are limited 
> if we have a misbehaving plugin

Was having both OTEL and misbehaving plugins in mind yes. Mostly OTEL (which could potentially allow us to measure performance as well?), but also things like passing a logger around, etc.